### ISSUE TITLE
Exchange message are flagged as readed when using POP3

### ISSUE THREAD

**Issue by asorcinelli** [jstedfast/MailKit#952]
Hello,
I'm trying to replace my current email library (Aspose) with Mailkit. All works fine, but I have a strange behaviour when I read messages from Exchange using POP3. When I use the old library the messages remains in the state "unread", while if I use Mailkit the messages are flaggeg as readed.
If I use other email accounts both libraries flag the messages as readed. I would have expected this behavior also on Exchange, since I read that the POP3 protocol should behave this way.

If this behavior is actually possible with the POP3 protocol, it would be interesting to be able to decide, via parameter, whether to flag the messages as read or not. In certain situations it is very convenient for the messages to remain in "unread" status.

**Comment by jstedfast** [comment]
POP3 doesn't support the concept of read vs unread, only IMAP does. Unfortunately this means that MailKit cannot influence the server's read/unread state via Pop3Client.

You could use MailKit's ImapClient, though.

**Comment by asorcinelli** [comment]
I know that POP3 doesn't support the concept of read and unread. I can't understand how the other library can leave messages unread on Exchange. Maybe Exchange have some special features? 
I didn't find any information about it. Could the log of commands that are sent to the Exchange POP3 server be useful?

**Comment by jstedfast** [comment]
It's possible Exchange has some extensions or something that allow the other library to leave messages marked unread on the server. If you can get me a log of the commands that the other client sends, then yes, it would be helpful - and perhaps I could implement whatever extension is being used.

**Comment by asorcinelli** [comment]
[Aspose.Email.Pop3_2019-11-19.log.zip](https://github.com/jstedfast/MailKit/files/3864812/Aspose.Email.Pop3_2019-11-19.log.zip)

here is the command log. I'm not an expert, but I can't see anything strange. But with these commands the messages remain in "unread" state on server.

**Comment by jstedfast** [comment]
It looks like Aspose is using the same commands that MailKit uses to retrieve messages (`RETR`) and according to the `CAPA` response at the beginning of the log, there are no extensions that I'm not familiar with (just `UIDL`, `SASL PLAIN` and `USER`).

That said, what strikes me is that with Aspose, after retrieving the last message, it seems to just drop the connection instead of logging out cleanly.

The way POP3 works with the `DELE` command (I know you aren't using `DELE`, but bear with me) is that the `DELE` command only marks a message for deletion (it's a bit like adding the `\Deleted` flag on IMAP) and those messages are only *actually* deleted if the client sends a `QUIT` command afterward. If the client drops the TCP/IP connection without calling `QUIT`, then the deleted flags are reset.

The same *may* be true for the read/unread flag.

In other words, you probably call `client.Disconnect (true);` in your MailKit code. Try calling `client.Disconnect (false);` instead to see if that solves it.

**Comment by asorcinelli** [comment]
It works! It is exactly as you say.
In addition to the DELE command, does QUIT command influence any other behavior? In my opinion Aspose dynamically manages the QUIT because if I delete messages from the server it sends it, if I don't do any modification it doesn't send it.

**Comment by jstedfast** [comment]
Not that I’m aware of. Only the deleted state is mentioned in the pop3 protocol specifications.

**Comment by asorcinelli** [comment]
ok. thank you very much!
